File: PTA/B_level/1007_PrimeGuess.py
# 素数对 
import math

def isPrime( N ):
	""" judge whether N is a prime number
    param:  N (int)
    retype: bool
    """
	if N <= 1:
		return False
	else:
		flag = True
		for  i in range(2, N):
			if (N % i) == 0:
				flag = False
				break
		return flag
        
def isPrimeSqrt( N ):
    if N <= 1:
        return False
    else:
        flag = True
        num = int( math.sqrt(N) )
        for  i in range(2, num+1):
            if (N % i) == 0:
                flag = False
                break
        return flag

def generatePrime( N ):
    prime = [ i for i in range(1,N+1) if isPrime(i) ]
    return prime
    
def sievePrime( n ):
    """ Implement the seive og Eratosthenes
    param:  n (int)
    retype: list(int)
    """
    A = [ i for i in range(n+1)]
    num = int( math.sqrt(n) )
    for p in range(2, num+1):
        if A[p] != 0: # p hasn't been eliminated on previous passes            
            j = p * p
            while j <= n: # p*p <= n
                A[j] = 0 # mark element as eliminated
                j = j + p # 剔除p的倍数
    L = [ item for item in A if item >= 2 ]
    return L

def seive( N ):
    if N < 2:
        return []
    else:
        L = [ i for i in range(2, N+1) ]
        for i in range(2, len(L)):
            L = list( filter( lambda x: (x%i != 0) or x == i , L) ) 
        return L
        
N = int( input() )
cntPrimePair = 0
differ = 0
#for i in N:
#primes = generatePrime( N )
primes = sievePrime( N )
length = len(primes)
for i in range(length):
    if i < (length - 1):
        differ = primes[i+1] - primes[i]
        if differ == 2:
            cntPrimePair += 1
print(cntPrimePair)
    

File: PTA/B_level/1007_PrimeGuess.cpp
#include <stdio.h>

int isPrime( int N )
{   /* 穷举法判断素数 */
	if ( N <= 1 ) {
		return 0;		
	} else {
		int flag = 1;
		for ( i = 2; i < N; i++ ) {
			if ( (N % i) == 0 ) {
				flag = 0;
                break;
			}
		}
		return flag;			
	}			
} 

int isPrimeSqrt( int num )
{   /* 开方法判断素数 */ 
    int flag = 1;
    int N = sqrt(num);
    int i;
    for ( i = 2; i <= N; i++ ){
        if (num % i == 0) {
            flag = 0;
        } 
    }
    return flag;
}

int generatePrime( int primes[], int N )
{
	int i, j;
	for ( j = 0, i = 1; i <= N; i++ ) {
		if ( isPrimeSqrt(i) ) {
			primes[j++] = i;
		}
	}
	return j; /* prime index */
}

int main(int argc, char *argv[])
{
	int i, N, border;
	int differ, cntPrimePair;
	border = differ = cntPrimePair = 0;
	scanf("%d", &N);
	
	int primes[N]; /* 存放素数 */ 
	for ( i = 0; i < N; i++ ) {
		primes[i] = 0;
	}
	border = generatePrime( primes, N );
	
	for ( i = 0; i < (border-1); i++ ) {
    	differ = primes[i+1] - primes[i];
    	//printf("differ %d\n", differ);
    	if ( differ == 2 ) {
        	cntPrimePair += 1;	    	
	    }
	}	
	printf("%d\n", cntPrimePair);
	 
    return 0;
}